### PR TITLE
DT-1649 Protect app with new role

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: server/authentication/roles.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { Response } from 'express'
+import jwtDecode from 'jwt-decode'
+
+const MAINTAINER_ROLE = 'HMPPS_REGISTERS_MAINTAINER'
+
+const extractRoles = (res: Response): Array<string> => {
+  const token = res?.locals?.user?.token
+  const decodedToken = token && (jwtDecode(res.locals.user.token) as { authorities?: string[] })
```
Comment (mikehalmamoj) [inline server/authentication/roles.ts:8]: Not sure what's going on with nullables here...?

**Review by Mjwillis — DISMISSED** [review]
Lgtm

File: server/authentication/roles.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { Response } from 'express'
+import jwtDecode from 'jwt-decode'
+
+const MAINTAINER_ROLE = 'HMPPS_REGISTERS_MAINTAINER'
+
+const extractRoles = (res: Response): Array<string> => {
```
Comment (mikehalmamoj) [inline server/authentication/roles.ts:6]: Is Array<String> the same as string[]?

File: integration_tests/mockApis/auth.js [hunk 3]
```diff
@@ -138,6 +138,7 @@ const stubUserRoles = () =>
 module.exports = {
   getLoginUrl,
   stubPing: () => Promise.all([ping(), tokenVerification.stubPing()]),
-  stubLogin: () => Promise.all([favicon(), redirect(), logout(), token(), tokenVerification.stubVerifyToken()]),
+  stubLogin: (roles = ['HMPPS_REGISTERS_MAINTAINER']) =>
```
Comment (mikehalmamoj) [inline integration_tests/mockApis/auth.js:141]: ROLE_?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**